pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -590,18 +587,122 @@ func (w *Webhook) newInjector(startTime time.Time, pod *corev1.Pod, opts ...inje
 		podMutator(w.version)
 }
 
+// podSumRessourceRequirements computes the sum of cpu/memory necessary for the whole pod.
+// This is computed as max(max(initContainer resources), sum(container resources)) for both limit and request
+func podSumRessourceRequirements(pod *corev1.Pod) corev1.ResourceRequirements {
+	ressourceRequirement := corev1.ResourceRequirements{
+		Limits:   corev1.ResourceList{},
+		Requests: corev1.ResourceList{},
+	}
+
+	for _, k := range [2]corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU} {
+		// Take max(initContainer ressource)
+		for i := range pod.Spec.InitContainers {
+			c := &pod.Spec.InitContainers[i]
+			if limit, ok := c.Resources.Limits[k]; ok {
+				existing := ressourceRequirement.Limits[k]
+				if limit.Cmp(existing) == 1 {
+					ressourceRequirement.Limits[k] = limit
+				}
+			}
+			if request, ok := c.Resources.Requests[k]; ok {
+				existing := ressourceRequirement.Requests[k]
+				if request.Cmp(existing) == 1 {
+					ressourceRequirement.Requests[k] = request
+				}
+			}
+		}
```

Comment:
I don't think we need to check initcontainer resources because they are run sequentially. 

However this function needs to check one kind of specific init-container that are now used as side card container: Since k8s 1.29 side car container can be defined as init-container but with a specific `restartPolicy` set to "Always"
see: https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/

Reply:
> I don't think we need to check initcontainer resources because they are run sequentially.

I believe we need to check them still. According to the link you sent 
```
The Pod's effective request/limit for a resource is [...] the higher of:
* the sum of all non-init containers(app and sidecar containers) request/limit for a resource
* the effective init request/limit for a resource
```

I updated this to sum sidecar container resources with normal container resources.